### ISSUE TITLE
Trust registration service

### ISSUE THREAD

**Issue by philsherry** [hmrc/accessibility#10]
✅ **Please tick the boxes below to show you’ve read the requirements.**

Before raising this request for an in-house accessibility audit, we did the following things as a team:

- [x] Made sure each page of our service contains valid HTML.
- [x] Made sure each page of our service is free from WCAG errors.
- [x] Tested each page of our service with at least one screen reader, to make sure it was usable.

**Comment by philsherry** [comment]
# Audit: trusts-relationship-establishment-frontend

## Executive summary

This service is currently failing.

There are basic errors and common issues on each page that would be picked up easily if simple guidelines were followed. Most of these are listed in the guidance that we recently published. Reading that will give you a good understanding of how to build more accessible code.

- <https://github.com/hmrc/accessibility>

## What is the lead trustee’s last name

- Form hints not tied to their respective inputs with `aria-describedby`, for example:
  ```html
  <div class="form-group">
    <h1 class="margin-bottom-10 margin-top-0">
      <label for="answer" class="heading-xlarge margin-bottom-20">
        What is the lead trustee’s last name?
      </label>
    </h1>
    <div class="form-group">
      <span class="form-hint margin-bottom-10" id="form-hint-identifier">
        This is the person who gets the official documents for the trust.
        For example, the notice to file tax returns.
      </span>
      <input id="answer" type="text" class="form-control" name="answer"
      value="" aria-invalid="false" aria-describedby="form-hint-identifier" />
    </div>
  </div>
  ```
- Error pages don't give focus to the error summary box, meaning the skip link steals it and the user is parked at the top of the page.
- Effort has been made to remove the focus outline from the error summary box. Why is this, and what other functionality did you remove at the same time? (See previous point.)
  ```css
  .flash.error-summary:focus {
    outline: none;
  }
  ```

### What is the lead trustee’s date of birth

- Date of birth component has many errors, all of which need fixing.

  `<input class="form-control" id="dateAnswer.day" type="number" class="form-control" size="2" maxlength="2" name="dateAnswer.day" value="">`
  - `maxlength` is only valid on `password`, `search`, `tel`, `text`, `url` input types.
  - `size` is only valid on `email`, `password`, `tel`, `text` input types.
  - `type="number"` tells the user it's incrementable text, but it doesn't work because of the invalid attributes.
  - `class` attribute appears twice, which is invalid.
  - Hint text isn't tied to the first input with `aria-describedby`.
  - Hint text doesn't tell the user which format to enter the date.

### What is the lead trustee’s passport number

- “What is the lead trustee’s passport number?” has no hint text at all. Isn't this only ever 9 digits, no matter the country? If so, it's safe to say that on there. Source: [Machine-readable passport](https://en.wikipedia.org/wiki/Machine-readable_passport).
- You should also use the correct input type for a reference which is made up of numbers.

### Check your answers before continuing

- Why are these pages using a `button` to `GET` the next page? Just use an anchor to link from one page to the next.
  ```html
  <form method="get" action="/check-your-identity-for-trusts/journey/entities/individual/0?redirectPage=interim-summary-page">
    <button id="submit-continue" class="button" type="submit">
      Continue
    </button>
  </form>
  ```

### What are the details of a trustee, settlor or beneficiary associated with the trust

- “This cannot be the lead trustee” should not be a heading. Use the appropriate markup and styling.
  `<p class="lede">This cannot be the lead trustee.</p>`
- “Date of birth” component has the same issues as above, but also…
- `label` for “Date of birth” wraps the entire component, which is invalid and causes many issues.

### Check your answers before continuing (Person associated with the trust)

- Why are these pages using a `button` to `GET` the next page? Just use an anchor to link from one page to the next.
  ```html
  <form method="get" action="/check-your-identity-for-trusts/journey/summary-page">
    <button id="submit-continue" class="button" type="submit">
      Continue
    </button>
  </form>
  ```

### Check your answers before you confirm them

- Checking them *is* confirming them. The text below says the following:
  > “Before you submit your answers, you must confirm they are correct.”
  The button also says Submit. So, I suggest changing the heading of this page to something less confusing, like “Check your answers before you submit them”.

### What is the lead trustee organisation’s UK postcode

- Postcode input field should be roughly the same width as the Continue button. [Use appropriately-sized text inputs](https://design-system.service.gov.uk/components/text-input/#use-appropriately-sized-text-inputs).

### What are the details of an organisation associated with the trust

- “They must match the details provided to HMRC when the trust was registered” should not be a heading. Use the appropriate markup and styling.
  `<p class="lede">They must match the details provided to HMRC when the trust was registered.</p>`
- Minor issues with the autocomplete: it doesn't show which option from the list I'm on if just using keyboard navigation. The image of a normal screen shows when I'm on an option, but it doesn't on the high contrast screen. Confirm you're using the latest version.
- ![autocomplete on a normal screen](https://user-images.githubusercontent.com/59015379/72975570-f4eabf00-3dc8-11ea-969c-2ae08851c3d6.png)
- ![autocomplete on a high contrast screen](https://user-images.githubusercontent.com/59015379/72975613-016f1780-3dc9-11ea-9e9e-06e2a9c65153.png)
- As this is the GDS accessible autocomplete, your team is taking it on faith that it's what it says it is—an accessible autocomplete. If you can fix it, great. Send a pull request to GDS so they can merge it in. Otherwise, raise an issue so they can investigate it.

## Queries

- `<input type="hidden" name="redirectPage" value="next-journey-step" aria-hidden="true">`
- `aria-invalid="false"` on a field which is asking for a surname?
- `aria-hidden="true"` on a hidden field? Unnecessary.
